src/util.c

```diff
@@ -49,6 +49,61 @@ char *path_append(const char *path, const char *filename)
     return str;
 }
 
+char *escape_char(const char *url, const ESCAPE_CHAR c)
+{
+    char escape_me;
+
+    /* A space, for example, becomes thrice bigger as '%20' after escaping */
+    int how_bigger = 3;
+
+    switch (c) {
+        case ESCAPED_CHAR_SPACE:
+            escape_me = ' ';
```

Comment:
I don't think this is a good idea, the characters that need to be escape is not going to change on the fly. You should define it globally, rather than creating a new type to wrap around it. I don't think we need this many indirection.